packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -28,10 +28,49 @@ export function PreviewIndicator({ children }) {
 }
 
 let buildId
+let pageData
+let latestCheckedBuild
 
 export default function Indicator() {
   const [buildInfo, setBuildInfo] = useState()
 
+  async function fetchPageData() {
+    const urlHostString = window.location.origin
+    const pathAdjustment =
+      window.location.pathname === `/` ? `/index` : window.location.pathname
+
+    const url = `${urlHostString}/page-data${pathAdjustment}/page-data.json`
+
+    const resp = await fetch(url)
+    const data = await resp.text()
+
+    return data
+  }
+
+  useEffect(() => {
+    // declare the data fetching function
+    const fetchData = async () => {
+      const data = await fetchPageData()
+      pageData = data
+    }
+
+    // call the function
+    fetchData()
+      // make sure to catch any error
+      .catch(console.error)
+  }, [])
```

Comment:
```suggestion
  useEffect(() => {
    try {
        const data = await fetchPageData()
        setPageData(data) 
    } catch(e) {
      console.error(e)
    }
  }, [])
```

I agree with Daniels comment that we should think about what we want to do if there's an error though 🤔